public/video-ui/src/services/UploadsApi.js

```diff
@@ -5,6 +5,18 @@ import { errorDetails } from '../util/errorDetails';
 import 'aws-sdk/dist/aws-sdk';
 const AWS = window.AWS;
 
+// The timeout for individual upload requests. Defaults to 120s. This
+// default causes problems on slow connections – for example, w/ a 0.5mbps
+// upload speed (3.75MB/minute), uploads for 8mb chunks will never complete.
+AWS.config.httpOptions.timeout = 240_000; // in ms
+const httpOptions = {
+  // The number of multipart uploads to run concurrently. Defaults to 4,
+  // which has caused problems with slow connections timing out requests
+  // prematurely. We judge allowing uploads on slow connections to be
+  // more valuable than a minor boost in upload speed due to concurrent uploads.
+  queueSize: 1
+};
```

Comment:
Is this two different ways of changing the same class of setting? The top a global override and the other per-call? If they could be done the same way then that might be less confusing?

Reply:
Agreed – sadly, looking at the type declaration for `.upload`, I don't think that's possible.

Reply:
That is to say, the HTTP settings seem global, and cannot be passed in any shape that I can find to `upload`, but hmm, having said that, we might be able to configure queueSize globally as you suggest! I'll take a look.

Reply:
I can't see a way to do this, but v. happy to be corrected. (Can only see `queueSize` refs in `ManagedUpload.ManagedUploadOptions`, which are only referenced in the signature for `upload`.)